### ISSUE TITLE
[Auth] Lambda trigger exception not forwarded when using social sign-in

### ISSUE THREAD

**Issue by dnys1** [aws-amplify/amplify-ios#1617]
### Describe the bug

https://github.com/aws-amplify/amplify-flutter/issues/1279

CX is not receiving exception message from Lambda trigger when using social sign-in. I was able to reproduce when using social sign-in only - regular sign-up and Hosted UI both returned the exception.

Is this expected behavior? And is there a way to expose these exceptions?

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Set up lambda trigger which throws an exception (I tested with pre-signup).
2. Set up social sign-in (I tested with Google).
3. Sign in with social login
4. Receive generic error message w/out Lambda exception.
```

### Expected behavior

For regular signup/Hosted UI, the message returned by the Lambda is included in the exception. For social sign-in, this information seems to be missing.

### Amplify Framework Version

1.15.5

### Amplify Categories

Auth

### Dependency manager

Cocoapods

### Swift version

5.0

### CLI version

7.6.10

### Xcode version

13.2

### Is this a regression?

No

### Device

iPhone 13 - SImulator

### iOS Version

iOS 15

**Comment by Sunsiha** [comment]
any update on this?

**Comment by ameter** [comment]
Thanks for bringing this to our attention.  We were able to reproduce this issue this morning.  The underlying issue is that lambda errors are handled differently by the service during regular signup/signIn, hostedUI, and social signIn, and in the case of social signIn, the error returned from the service is not being properly handled by Amplify.  We are investigating the best way to handle this with the information we receive from the service and will provide an update here when we have more information.

**Comment by ameter** [comment]
I've created [PR](https://github.com/aws-amplify/amplify-ios/pull/1642) to address this issue.  Note that the errors returned will be service errors instead of lambda errors due to the fact that the Social SignIn service lumps multiple types of errors together.  However, these errors will now be properly handled and the error description will now include the error message from the Lambda.

@dnys1 @Sunsiha Please feel free to test with [this branch](https://github.com/aws-amplify/amplify-ios/tree/fix/lambdaErrorAuth).

**Comment by ameter** [comment]
This fix was released in Amplify iOS version 1.21.0.

**Comment by AsitDixit** [comment]
@ameter 

> This fix was released in Amplify iOS version 1.21.0.

When in Android??

**Comment by ruisebas** [comment]
@AsitDixit I see you already created an issue in Android's repo: https://github.com/aws-amplify/amplify-android/issues/1907
Please follow up there.
